### ISSUE TITLE
url provider should force http/https

### ISSUE THREAD

**Issue by ludoo0d0a** [leaflet-extras/leaflet-providers#227]
I try to use your plugin in chrome extension.
Since url templates looks like url: '//{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', it failed to load tiles because under chrome:// protocol

I suppose your template is to stay in the same protocol as the website, but then can you add a test on current location.protocol and add a fallback as http ?

``` js
var protocol = /https?/.test(location.protocol)?location.protocol:'http';
```

**Comment by jieter** [comment]
Hmm, makes sense to whitelist only `http` and `https`, but should we fall back to http or https? Thoughs @brunob?

**Comment by brunob** [comment]
@jieter ok, we could do something inspired by what i do in bing layer plugin, fall back to http when the page is on file:// protocol see : 

https://github.com/shramov/leaflet-plugins/blob/master/layer/tile/Bing.js#L54

**Comment by ludoo0d0a** [comment]
warning : as I mentioned, in Chrome extension scope, this is chrome:// protocol, not file:// !

**Comment by brunob** [comment]
@ludoo0d0a if you look at the code i'm pointing, you'll see that it use http only if the protocol of the page is file, in the other cases it stay in the document protocol.

**Comment by ludoo0d0a** [comment]
yes, that's the problem ;)

Here the result in console in a chrome extension...
```js
>(document.location.protocol === 'file:') ? 'http' : document.location.protocol.slice(0, -1);
>"chrome-extension"
```
if urlScheme stay the same as document.location.protocol, you never could reach our remote content...

note protocol is chrome-extension:// in extension and chrome:// in chrome app (from what i remember)

**Comment by brunob** [comment]
Ha my bad, so we just have to adapt the test a bit. Anyway, i was pointing this code in order to not forget users who will try to use the script locally (since https by default will break this use case).

**Comment by ludoo0d0a** [comment]
pas de souci :stuck_out_tongue_winking_eye:

**Comment by brunob** [comment]
@ludoo0d0a do you have a patch to propose or should we get on this ?

**Comment by brunob** [comment]
Fixed by #258
